### PR TITLE
Blogapp ui changes2

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
# Blog App Frontend:
Description: There are the following changes added:
1. Slug url for blog post detail page
2. Delete Confirmation Modal added
3. Style changes on the respective pages.

# Installation Instructions:
1. run `npm install`
2. run npm start

Screenshot from 2019-10-01 16-39-32
Screenshot from 2019-10-01 16-39-44
Screenshot from 2019-10-01 16-40-35
Screenshot from 2019-10-01 16-40-56
Screenshot from 2019-10-01 16-41-22

### CODE REVIEW

File: src/actions/blogList.js [hunk 1]
```diff
@@ -10,9 +10,9 @@ export const blogUpvote = (id) => (dispatch, getState) => {
     axios.get(serverData.django_server + `/api/blog/${id}/upvote/`, tokenConfig(getState))
         .then(res => {
             if (res.data === "already_voted")
-                alert("You've already voted!");
+                dispatch(createMessage({vote: "You have already Up-Voted"}));
```
Comment (zubair-arbi) [inline src/actions/blogList.js]: We are now sending this message from the backend.

File: src/components/accounts/Login.js [hunk 2]
```diff
@@ -34,8 +33,12 @@ export class Login extends Component {
                 <div className="card card-body mt-5">
                     <h2 className="text-center">Login</h2>
                     <form onSubmit={this.onSubmit}>
+                        {this.props.errors.msg.non_field_errors ?
+                            <span className='alert alert-danger'>{this.props.errors.msg.non_field_errors}</span>
+                            : null}
+
                         <div className="form-group">
-                            <label>Username</label>
+                            <label className="mt-2">Username*</label>
```
Comment (zubair-arbi) [inline src/components/accounts/Login.js]: Here are accessible ways to make a field required: https://dequeuniversity.com/tips/identify-required-elements
Use any of these.

Helpful link: https://www.w3.org/TR/WCAG20-TECHS/ARIA2.html

File: src/components/blog/BlogItem.js [hunk 3]
```diff
@@ -133,108 +126,171 @@ class BlogItem extends Component {
...
+                                 alt=""
+                            />
+                            :
+                            <img className="mt-4 img-fluid rounded"
+                                 src={defaultImage}
+                                 alt=""
+                            />
+                        }
+
+                        <br/><br/><br/>
```
Comment (zubair-arbi) [inline src/components/blog/BlogItem.js]: Avoid doing this, use one `<br/>` and add padding/margin on the elements before and after this

**Review by zubair-arbi — CHANGES_REQUESTED** [review]

File: src/actions/blogList.js (hunk 1 above)
Comment (Ali-D-Akbar) [reply src/actions/blogList.js]: It's been changed

File: src/components/accounts/Login.js (hunk 2 above)
Comment (Ali-D-Akbar) [reply src/components/accounts/Login.js]: I've added this:
<p id="formInstructions">Fields marked with an asterisk (*) are required.</p>

File: src/components/blog/BlogItem.js (hunk 3 above)
Comment (Ali-D-Akbar) [reply src/components/blog/BlogItem.js]: It's been removed.

**Comment by zubair-arbi** [comment]
Please merge the https://github.com/Ali-D-Akbar/react-redux-blog-app-frontend/pull/7 then rebase this PR onto master.

**Comment by Ali-D-Akbar** [comment]
PR #7 has been merged and applied rebase PR #8 onto master.

**Pull request merged**